src/supermarket/app/views/api/v1/cookbook_versions/_cookbook_version.json.jbuilder

```diff
@@ -4,14 +4,16 @@ json.version cookbook_version.version
 json.average_rating nil
 json.cookbook api_v1_cookbook_url(cookbook)
 json.file api_v1_cookbook_version_download_url(cookbook_version.cookbook, cookbook_version)
-json.quality do
-  json.foodcritic do
-    json.failed @cookbook.foodcritic_failure
-    json.feedback @cookbook.foodcritic_feedback
-  end
-  json.collaborator do
-    json.failed @cookbook.collaborator_failure
-    json.feedback @cookbook.collaborator_feedback
+if cookbook_version.metric_results.any?
+  json.quality do
+    json.foodcritic do
+      json.failed foodcritic_metric_result(cookbook_version).failure
+      json.feedback foodcritic_metric_result(cookbook_version).feedback
+    end
+    json.collaborator do
+      json.failed collaborator_num_metric_result(cookbook_version).failure
+      json.feedback collaborator_num_metric_result(cookbook_version).feedback
```

Comment:
The [change](https://github.com/chef/supermarket/commit/ea91d0d80462306f7526735ee231bf4cd5413c9a) ...

```diff
-if cookbook_version.metric_results.any?
+if @cookbook_version_metrics.any?
```

... helps. But  I think the call to `find_by` in [foodcritic_metric_result](https://github.com/chef/supermarket/pull/1433/files#diff-44a50391e9501944d79581f30b09ddcdR76) and [collaborator_num_metric_result](https://github.com/chef/supermarket/pull/1433/files#diff-44a50391e9501944d79581f30b09ddcdR88) will do more look-up queries.